src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -583,17 +613,46 @@ private static Site prepareSite(Site site, ExtSite extSite, boolean useFirstPart
                 : site;
     }
 
+    /**
+     * Make Source with corresponding request.ext.prebid.schains
+     */
+    private static Source prepareSource(String bidder, Map<String, ObjectNode> bidderToSchain, Source receivedSource) {
+        final ObjectNode defaultSchain = bidderToSchain.get(GENERIC_SCHAIN_KEY);
+        final ObjectNode bidderSchain = bidderToSchain.getOrDefault(bidder, defaultSchain);
+
+        if (bidderSchain == null || bidderSchain.isNull()) {
+            return receivedSource;
+        }
+
+        final ObjectNode jsonExtSource = Json.mapper.valueToTree(ExtSource.of(bidderSchain));
+
+        if (receivedSource == null) {
+            return Source.builder().ext(jsonExtSource).build();
+        } else {
+            return receivedSource.toBuilder().ext(jsonExtSource).build();
+        }
+    }
+
     /**
      * Removes all bidders except the given bidder from bidrequest.ext.prebid.data.bidders and
      * bidrequest.ext.prebid.bidders to hide list of allowed bidders from initial request.
+     * Also mask bidrequest.ext.prebid.schains.
      */
     private static ObjectNode prepareExt(String bidder, List<String> firstPartyDataBidders,
                                          Map<String, JsonNode> bidderToPrebidBidders, ExtBidRequest requestExt,
                                          ObjectNode requestExtNode) {
-        if (firstPartyDataBidders.isEmpty() && bidderToPrebidBidders.isEmpty()) {
+        final ExtRequestPrebid extPrebid = requestExt != null ? requestExt.getPrebid() : null;
+        final List<ExtRequestPrebidSchain> extPrebidSchains = extPrebid != null ? extPrebid.getSchains() : null;
+        final boolean suppressSchains = extPrebidSchains != null;
+
+        if (firstPartyDataBidders.isEmpty() && bidderToPrebidBidders.isEmpty() && !suppressSchains) {
             return requestExtNode;
         }
 
+        final ExtRequestPrebid.ExtRequestPrebidBuilder extPrebidBuilder = extPrebid != null
```

Comment:
I would suggest moving this declaration closer to its usage, i.e. right before `return Json.mapper.valueToTree(ExtBidRequest.of(extPrebidBuilder`